### ISSUE TITLE
Floating point type

### ISSUE THREAD

**Issue by nomeata** [llvm-hs/llvm-hs-typed#1]
Maybe this is a discussion that should be had in `llvm-hs-pure`…

The [LLVM language reference](releases.llvm.org/4.0.0/docs/LangRef.html#floating-point-types) specifies the following floating point types:

Type | Description
----- | ---------------
half |	16-bit floating point value
float |	32-bit floating point value
double | 	64-bit floating point value
fp128 |	128-bit floating point value (112-bit mantissa)
x86_fp80 | 	80-bit floating point value (X87)
ppc_fp128 |	128-bit floating point value (two 64-bits)

Why does the llvm-hs AST have this complicated
```
  | FloatingPointType { typeBits :: Word32, floatingPointFormat :: FloatingPointFormat }
```
constructor? The values (`SomeFloat` in `LLVM.AST.Float`) follow the LLVM spec more closely…

And, more specific to this repository: Shall we stick to the type that llvm-hs uses, or shall we use something like this:
```
data FloatingPointType = HalfType | FloatType | DoubleType | FP128Type | QuadrupleType | X86_FP80Type | PPC_FP128Type
…
  | FloatingPointType { floatingPointType = FloatingPointType }
```

**Comment by sdiehl** [comment]
The answer for this is almost certainly because it was inherited from the old `llvm-general-pure` codebase.

The later makes more sense for lifting to the typelevel since most code is using standard sized types.

**Comment by nomeata** [comment]
Fixed in dda2392.
